### PR TITLE
Refactoring of compas_robots to COMPAS 2.0 and various updates to prepare for release

### PR DESCRIPTION
**Pull request by gonzalocasas**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by gonzalocasas** [comment]
ah wait.. hold on the review, the build is failing horribly for some reason...

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-compas-dev
+Copyright (c) 2023 COMPAS Association
```
Comment (chenkasirer) [inline LICENSE]: haha just out of curiosity, are we expected to update this every year? what does this even mean?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+compas @ git+https://github.com/compas-dev/compas@main
```
Comment (chenkasirer) [inline requirements.txt:1]: If I remember right, Pypi don't accept git requirements.. maybe pin to beta?

File: src/compas_robots/blender/scene/robotmodelobject.py [hunk 3]
```diff
@@ -1,44 +1,39 @@
...
-
-from compas_robots.artists import RobotModelArtist
-from compas_robots.robots import RobotModel
+from compas_robots import RobotModel
+from compas_robots.scene import BaseRobotModelObject
 
 
-class RobotModelArtist(BlenderArtist, RobotModelArtist):
-    """Artist for drawing robot models in Blender.
+class RobotModelObject(BlenderSceneObject, BaseRobotModelObject):
```
Comment (chenkasirer) [inline src/compas_robots/blender/scene/robotmodelobject.py:18]: mmh `RobotModelObject` sounds a bit too generic to me `RobotModelSceneObject` too long/verbose?

File: src/compas_robots/configuration.py [hunk 4]
```diff
@@ -343,25 +343,24 @@ def from_prismatic_and_revolute_values(cls, prismatic_values, revolute_values, j
         )
 
     @property
-    def data(self):
-        """dict : The data representing the configuration.
+    def dtype(self):
+        return "compas_robots/Configuration"
```
Comment (chenkasirer) [inline src/compas_robots/configuration.py:347]: ah is this because we normally assume 2nd level import?

File: src/compas_robots/configuration.py [hunk 5]
```diff
@@ -343,25 +343,24 @@ def from_prismatic_and_revolute_values(cls, prismatic_values, revolute_values, j
...
-    def data(self, data):
-        self._joint_values = FixedLengthList(data.get("joint_values") or data.get("values") or [])
-        self._joint_types = FixedLengthList(data.get("joint_types") or data.get("types") or [])
-        self._joint_names = FixedLengthList(data.get("joint_names") or [])
+    @classmethod
+    def from_data(cls, data):
+        return cls(
+            data.get("joint_values") or data.get("values") or [],
+            data.get("joint_types") or data.get("types") or [],
+            data.get("joint_names") or [],
```
Comment (chenkasirer) [inline src/compas_robots/configuration.py:362]: why would there be `values` or `types` keys?

File: src/compas_robots/model/geometry.py [hunk 6]
```diff
@@ -308,64 +260,39 @@ def data(self):
...
+        if self.color:
+            return Color(*self.color.rgba)
 
-#     # The current scenario is that we need to figure out the object type based on the DATASCHEMA
-#     keys = tuple(sorted(data.keys()))
-#     return TYPE_CLASS_ENUM_BY_DATA[keys]
+        return None
 
 
 class Geometry(Data):
```
Comment (chenkasirer) [inline src/compas_robots/model/geometry.py:298]: I know this isn't new but given that there's a `Geometry` in core maybe we should at some point rename this to `LinkGeometry`?

File: src/compas_robots/model/joint.py [hunk 7]
```diff
@@ -100,11 +93,9 @@ def data(self):
...
         }
 
-    @data.setter
-    def data(self, data):
-        self.rising = data["rising"]
-        self.falling = data["falling"]
-        self.reference_position = data["reference_position"]
+    @classmethod
+    def from_data(cls, data):
+        return cls(data["rising"], data["falling"], data["reference_position"])
```
Comment (chenkasirer) [inline src/compas_robots/model/joint.py:98]: I think here it's safe to omit `from_data` as the `data` dict matches signature of `__init__()`

File: src/compas_robots/model/joint.py [hunk 8]
```diff
@@ -234,17 +219,9 @@ def data(self):
...
-        self.joint = data["joint"]
-        self.multiplier = data["multiplier"]
-        self.offset = data["offset"]
-
     @classmethod
     def from_data(cls, data):
-        mimic = cls(data["joint"])
-        mimic.data = data
-        return mimic
+        return cls(data["joint"], data["multiplier"], data["offset"])
```
Comment (chenkasirer) [inline src/compas_robots/model/joint.py:224]: I think can be omitted

File: src/compas_robots/model/joint.py [hunk 9]
```diff
@@ -279,18 +256,9 @@ def data(self):
...
-        self.k_position = data["k_position"]
-        self.soft_lower_limit = data["soft_lower_limit"]
-        self.soft_upper_limit = data["soft_upper_limit"]
-
     @classmethod
     def from_data(cls, data):
-        sc = cls(data["k_velocity"])
-        sc.data = data
-        return sc
+        return cls(data["k_velocity"], data["k_position"], data["soft_lower_limit"], data["soft_upper_limit"])
```
Comment (chenkasirer) [inline src/compas_robots/model/joint.py:261]: I think can be omitted

File: src/compas_robots/model/link.py [hunk 10]
```diff
@@ -45,13 +46,9 @@ def get_urdf_element(self):
...
         return {"value": self.value}
 
-    @data.setter
-    def data(self, data):
-        self.value = data["value"]
-
     @classmethod
     def from_data(cls, data):
-        return cls(**data)
+        return cls(data["value"])
```
Comment (chenkasirer) [inline src/compas_robots/model/link.py:51]: I think can be omitted

**Review by chenkasirer — APPROVED** [review]
LGTM + left some comments

File: LICENSE (hunk 1 above)
Comment (gonzalocasas) [reply LICENSE]: Yes, I think we are. There are ever github actions for this: https://github.com/marketplace/actions/update-license-copyright-year-s

Comment (gonzalocasas) [reply LICENSE]: I'm not entirely certain about the legal implications of claiming copyright every year, I guess it's related to the expiration of copyright.

File: requirements.txt (hunk 2 above)
Comment (gonzalocasas) [reply requirements.txt:1]: yep, this is only temporary, before release, we need to fix this

File: src/compas_robots/blender/scene/robotmodelobject.py (hunk 3 above)
Comment (gonzalocasas) [reply src/compas_robots/blender/scene/robotmodelobject.py:18]: I had it like that at first, but the convention in core seems to be `XxxObject` not `XxxSceneObject` (eg `BoxObject`, `NetworkObject`, etc)

File: requirements.txt (hunk 2 above)
Comment (gonzalocasas) [reply requirements.txt:1]: https://github.com/compas-dev/compas_robots/issues/3

File: src/compas_robots/configuration.py (hunk 4 above)
Comment (gonzalocasas) [reply src/compas_robots/configuration.py:348]: yep, exactly

File: src/compas_robots/configuration.py (hunk 5 above)
Comment (gonzalocasas) [reply src/compas_robots/configuration.py]: Good catch, this is legacy from a very early version of compas_fab. I removed it at https://github.com/compas-dev/compas_robots/pull/1/commits/e6bc719ea9724436cdd11a64f5937255e58f02c4

File: src/compas_robots/model/geometry.py (hunk 6 above)
Comment (gonzalocasas) [reply src/compas_robots/model/geometry.py]: Good idea! Renamed at https://github.com/compas-dev/compas_robots/pull/1/commits/177312b75e81e2018d1dcde234f4778cb38bc402

File: src/compas_robots/model/joint.py (hunk 7 above)
Comment (gonzalocasas) [reply src/compas_robots/model/joint.py]: True, removed all redundant `from_data` in commit https://github.com/compas-dev/compas_robots/pull/1/commits/db0b3d35685a384ac9c30e493add6ba8e1e9945b

File: src/compas_robots/configuration.py (hunk 5 above)
Comment (chenkasirer) [reply src/compas_robots/configuration.py]: then I guess this `from_data` can also be removed :)

**Review by chenkasirer — APPROVED** [review]
LGTM!

**Pull request merged**